test/unit/errors/ignore.test.js

```diff
@@ -5,223 +5,270 @@
-        expect(otherErrorMetric).to.not.exist
-      })
+      t.end()
     })
+  })
+
+  t.test('Ignore status code should result in 0 errors reported', (t) => {
+    helper.runInTransaction(agent, function(tx) {
+      const errorAggr = agent.errors
+      agent.config.error_collector.capture_events = true
+      agent.config.error_collector.ignore_status_codes = [500]
+      tx.statusCode = 500
 
-    it('Ignore messages should trump expected_messages', function() {
-      helper.runInTransaction(agent, function(tx) {
-        const errorAggr = agent.errors
-        agent.config.error_collector.capture_events = true
-        agent.config.error_collector.ignore_messages = {"Error":['ignore']}
-        agent.config.error_collector.expected_messages = {"Error":['ignore']}
+      const error1 = new Error('ignore')
+      const error2 = new Error('ignore me too')
+      const error3 = new ReferenceError('i will also be ignored')
 
-        const error1 = new Error('ignore')
-        const error2 = new Error('not ignore')
-        const error3 = new ReferenceError('not ignore')
+      errorAggr.add(tx, error1)
+      errorAggr.add(tx, error2)
+      errorAggr.add(tx, error3)
 
-        errorAggr.add(tx, error1)
-        errorAggr.add(tx, error2)
-        errorAggr.add(tx, error3)
+      tx.end()
 
-        tx.end()
+      t.equal(errorAggr.traceAggregator.errors.length, 0)
 
-        expect(errorAggr.traceAggregator.errors.length).equals(2)
+      const transactionErrorMetric
+        = agent.metrics.getMetric(NAMES.ERRORS.PREFIX + tx.getFullName())
 
-        const transactionErrorMetric
-          = agent.metrics.getMetric(NAMES.ERRORS.PREFIX + tx.getFullName())
+      const allErrorMetric = agent.metrics.getMetric(NAMES.ERRORS.ALL)
+      const webErrorMetric = agent.metrics.getMetric(NAMES.ERRORS.WEB)
+      const otherErrorMetric = agent.metrics.getMetric(NAMES.ERRORS.OTHER)
 
-        const allErrorMetric = agent.metrics.getMetric(NAMES.ERRORS.ALL)
-        const webErrorMetric = agent.metrics.getMetric(NAMES.ERRORS.WEB)
-        const otherErrorMetric = agent.metrics.getMetric(NAMES.ERRORS.OTHER)
+      t.notOk(transactionErrorMetric)
 
-        expect(transactionErrorMetric.callCount).equals(2)
+      t.notOk(allErrorMetric)
+      t.notOk(webErrorMetric)
+      t.notOk(otherErrorMetric)
 
-        expect(allErrorMetric.callCount).equals(2)
-        expect(webErrorMetric.callCount).equals(2)
-        expect(otherErrorMetric).to.not.exist
-      })
+      t.end()
     })
+  })
 
-    it('Ignore status code should result in 0 errors reported', function() {
-      helper.runInTransaction(agent, function(tx) {
-        const errorAggr = agent.errors
-        agent.config.error_collector.capture_events = true
-        agent.config.error_collector.ignore_status_codes = [500]
-        tx.statusCode = 500
+  t.test('Ignore status code should ignore when status set after collecting errors', (t) => {
```

Comment:
The unified diff makes it hard to see... but this is the new test. It will be easier to see by looking at the commit diffs instead of the full one. I separated out the tap conversion into its own commit to keep cleaner.